images/base/files/usr/local/bin/entrypoint

```diff
@@ -405,15 +405,49 @@ enable_network_magic(){
   fi
 
   # patch docker's iptables rules to switch out the DNS IP
-  iptables-save \
-    | sed \
-      `# switch docker DNS DNAT rules to our chosen IP` \
-      -e "s/-d ${docker_embedded_dns_ip}/-d ${docker_host_ip}/g" \
-      `# we need to also apply these rules to non-local traffic (from pods)` \
-      -e 's/-A OUTPUT \(.*\) -j DOCKER_OUTPUT/\0\n-A PREROUTING \1 -j DOCKER_OUTPUT/' \
-      `# switch docker DNS SNAT rules rules to our chosen IP` \
-      -e "s/--to-source :53/--to-source ${docker_host_ip}:53/g"\
-    | iptables-restore
+  # we can't use iptables-save | (sed) | iptables-restore because is not safe (#3054)
+  local iptables_original
+  iptables_original="$(iptables-save -t nat | grep DOCKER)"
```

Comment:
we only care about NAT rules, don't we?

Reply:
I think so, also this is only relevant in the case of docker with embedded DNS.
We can skip this for podman.

I dislike how complex this is getting.